pypeit/wavetilts.py

```diff
@@ -299,11 +318,138 @@ def trace_tilts(self, arcimg, lines_spec, lines_spat, thismask, slit_cen):
                                            spat_order=self.par['spat_order'],
                                            maxdev_tracefit=self.par['maxdev_tracefit'],
                                            sigrej_trace=self.par['sigrej_trace'])
+                                           #, debug_pca=True, show_tracefits=True)
 
         # Return
         self.steps.append(inspect.stack()[0][3])
         return trace_dict
 
+    def model_arc_continuum(self, debug=False):
+        """
+        Model the continuum of the arc image.
+
+        The method uses the arc spectra extracted using
+        :attr:`extract_arcs` and fits a characteristic low-order
+        continuum for each slit/order using
+        :func:`pypeit.util.robust_polyfit_djs` and the parameters
+        `cont_function`, `cont_order`, and `cont_rej` from
+        :attr:`par`. The characteristic continuum is then rescaled to
+        match the continuum at each spatial position in the
+        slit/order.
+        
+        .. note::
+            The approach used here may be too simplistic (in the
+            robustness of the continuum fit and then how the
+            continuum is rescaled and projected for each slit/order).
+            Tests should be performed to make sure that the approach
+            is good enough to trace the centroid of the arc/sky lines
+            without biasing the centroids due to a non-zero continuum
+            level.
+
+        Args:
+            debug (:obj:`bool`, optional):
+                Run the method in debug mode.
+
+        Returns:
+            numpy.ndarray: Returns a 2D image with the same shape as
+            :attr:`msarc` with the model continuum.
+        """
+        # TODO: Instead check that extract arcs has been run using the
+        # "steps" attribute?
+        if self.arccen is None:
+            # Extract the arc spectra for all slits
+            self.arccen, self.arccen_bpm, self.arc_maskslit = self.extract_arcs()
+
+        # TODO: Should make this operation part of WaveTiltsPar ...
+        # Parse the upper and lower sigma rejection thresholds; used
+        # when rescaling continuum from center spectrum.
+        lower_rej, upper_rej = self.par['cont_rej'] if hasattr(self.par['cont_rej'], '__len__') \
+                                    else numpy.repeat(self.par['cont_rej'], 2)
+
+        # Fit the continuum of the extracted arc spectra for each slit
+        nspec, nslits = self.arccen.shape
+        spec = np.arange(nspec, dtype=float)
+        arc_continuum = np.zeros(self.arccen.shape, dtype=float)
+        arc_fitmask = np.zeros(self.arccen.shape, dtype=bool)
+        for i in range(nslits):
+            # TODO: What to do with the following iter_continuum parameters?:
```

Comment:
I would check what we use for the weavelength soulutions in wavecal. I think those may even be in a parset.

Reply:
That's what this is currently doing, I think.